src/tcgwars/logic/impl/gen4/CallOfLegends.groovy

```diff
@@ -1593,110 +788,80 @@ public enum CallOfLegends implements LogicCardInfo {
+        return copy(HeartgoldSoulsilver.VULPIX_87, this);
       case CHEERLEADER_S_CHEER_76:
       return copy(Unleashed.CHEERLEADER_S_CHEER_71, this);
       case COPYCAT_77:
-        return basicTrainer (this) {
-          text "You can play only one Supporter card each turn. When you play this card, put it next to your Active Pokémon. When your turn ends, discard this card.\nShuffle your hand into your deck. Then, draw a number of cards equal to the number of cards in your opponent’s hand."
-          onPlay {
-          }
-          playRequirement{
-          }
-        };
+        return copy(TeamRocketReturns.COPYCAT_83, this);
       case DUAL_BALL_78:
-        return basicTrainer (this) {
-          text "Flip 2 coins. For each heads, search your deck for a Basic Pokémon, show it to your opponent, and put it into your hand. If you do, shuffle your deck afterward."
-          onPlay {
-          }
-          playRequirement{
-          }
-        };
+        return copy(DeltaSpecies.DUAL_BALL_89, this);
       case INTERVIEWER_S_QUESTIONS_79:
-        return basicTrainer (this) {
-          text "You can play only one Supporter card each turn. When you play this card, put it next to your Active Pokémon. When your turn ends, discard this card.\nLook at the top 8 cards of your deck. Choose as many Energy cards as you like, show them to your opponent, and put them into your hand. Shuffle the other cards back into your deck."
-          onPlay {
-          }
-          playRequirement{
-          }
-        };
+        return copy(Unleashed.INTERVIEWER_S_QUESTIONS_77, this)
       case LOST_REMOVER_80:
         return basicTrainer (this) {
           text "Put 1 Special Energy card attached to 1 of your opponent’s Pokémon in the Lost Zone."
           onPlay {
+            def pcs = opp.all.findAll{it.cards.filterByType(SPECIAL_ENERGY)}.select("Remove special energy from which Pokémon?")
+            pcs.cards.filterByTYpe(SPECIAL_ENERGY).select("Put which special energy in the Lost Zone?").first().moveTo(opp.lostZone)
           }
           playRequirement{
+            assert opp.all.find{it.cards.filterByType(SPECIAL_ENERGY)}
           }
         };
       case LOST_WORLD_81:
-        return basicTrainer (this) {
+        return stadium (this) {
           text "This card stays in play when you play it. Discard this card if another Stadium card comes into play. If another card with the same name is in play, you can’t play this card.\nOnce during each player’s turn, if this player’s opponent has 6 or more Pokémon in the Lost Zone, the player may choose to win the game."
+          def lastTurn=0
+          def actions=[]
           onPlay {
+            actions=action("Stadium: Lost World") {
+              assert lastTurn != bg().turnCount : "You've already used Lost World this turn."
+              bc "Used Lost World."
+              lastTurn = bg().turnCount
+              bg.getGame().endGame(self.owner, WinCondition.OTHER);
```

Comment:
A stadium card does not have a `self`. I think you may want to do something with the `getStadiumInfoStruct` here?
```suggestion
              bg.getGame().endGame(bg.getStadiumInfoStruct().stadiumCard.player, WinCondition.OTHER);
```